tests/ui/no_mangle_with_rust_abi.rs

```diff
@@ -1,6 +1,21 @@
+//@aux-build:no_mangle_with_rust_abi.rs
 #![allow(unused)]
 #![warn(clippy::no_mangle_with_rust_abi)]
 
+extern crate no_mangle_with_rust_abi as external;
+
+pub struct UsingMeInCIsUB(u32, u32);
+#[no_mangle]
+pub static ZERO_UB: UsingMeInCIsUB = UsingMeInCIsUB(0, 0);
+
+#[repr(C)]
+pub struct UsingMeInCIsFine(u32, u32);
+#[no_mangle]
+pub static ZERO_DB: UsingMeInCIsFine = UsingMeInCIsFine(0, 0);
+
+#[no_mangle]
+pub static ZERO_UB_AGAIN: external::UsingMeInCIsUB = external::UsingMeInCIsUB(0, 0);
+
```

Comment:
Would be good to have a test for a static pointer/primitive

Reply:
Pointers will require a wrapper type since they are `!Send` + `!Sync`, and it only lints ADTs already, so I *think* that should work ok.

Reply:
Oh yeah forgot that about pointers, was looking at some pre 1.0 code apparently. Still want a test for some other primitive though, yeah the implementation looks good for them